tests/integration/security/citadel/namespace_targeting_test.go

```diff
@@ -0,0 +1,86 @@
+package citadel
+
+import (
+	"fmt"
+	"istio.io/istio/pkg/test"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	v1 "k8s.io/api/core/v1"
+	mv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/citadel"
+)
+
+const (
+	testNsPrefix       = "targeting-test"
+	serviceAccountName = "sample-service-account"
+)
+
+// TestNamespaceTargeting verifies that Citadel respects the designated namespace targeting labels.
+// current implementation is closely tied with Kubernetes concepts, but this is difficult to avoid
+// until Citadel itself is less closely tied with Kubernetes concepts
+func TestNamespaceTargeting(t *testing.T) {
+	framework.NewTest(t).Run(func(ctx framework.TestContext) {
+
+		citadel.NewOrFail(t, ctx, citadel.Config{Istio: ist})
```

Comment:
You don't need to create an extra Citadel here, the test framework deploys Istio control plane with Citadel already.